lib/date_range.rb

```diff
@@ -0,0 +1,35 @@
+module Hotel
+  class DateRange
+    attr_reader :start_date, :end_date 
+    attr_accessor :dates
+
+    def initialize(start_date:, end_date:)
+      @start_date = start_date
+      @end_date = end_date
+      @dates = Array(@start_date .. @end_date)
+    
+      # User: I want an exception raised when an invalid date range is provided, so that I can't make a reservation for an invalid date range
+      if @end_date < @start_date || @start_date == @end_date
+        raise ArgumentError.new("Your check-out date must be after your check-in.")
```

Comment:
It's helpful to include the bad arguments in your error messages:

```suggestion
        raise ArgumentError.new("Your check-out date (#{@start_date}) must be after your check-in (#{end_date}).")
```